### PR TITLE
Add the top section of the landing page

### PR DESCRIPTION
**Pull request by gracelungu**
#### Description of changes made

This PR adds the top section of the landing page

#### Type of change

> **Please select the relevant options**

- [x] New feature (non-breaking changes which add the new functionalities)
- [x] Bug fix (non-breaking changes which fix an issue)
- [ ] Documentation (added or improved the documentation)
- [ ] Breaking change (fix or a feature that would cause existing feature not to work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

> **Please describe the tests that you ran to verify your changes**

- After cloning the [repo](https://github.com/kisanola/AlphaCodeFrontend.git)
- Check out this branch by `landing-top-section`
- Run `yarn or npm install` to install the project dependencies
- Run `yarn start`

- [ ] Unit
- [ ] Integration
- [ ] End to end

#### Add the link to the to gitKraken board?

[Story on the board](https://app.gitkraken.com/glo/card/1fe1d836549c4d7cac4b3444c455e784)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have ensured coding conventions for my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests which prove the effectiveness of my work
- [ ] New and existing tests pass locally with my changes

#### Screenshot
image

### CODE REVIEW

**Review by espoirMur — COMMENTED** [review]
LGTM, But where are tests?

File: package.json [hunk 1]
```diff
@@ -10,18 +10,23 @@
     "@types/node": "^12.0.0",
     "@types/react": "^16.9.0",
     "@types/react-dom": "^16.9.0",
+    "@types/react-router-dom": "^5.1.3",
```
Comment (oesukam) [inline package.json]: This should go in the dev section plus all other type definitions to reduce on the size of the build file

File: src/Components/Header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React, { Component } from 'react';
+import './Header.scss';
+
+class Header extends Component {
+    render() {
+        return (
+            <React.Fragment>
```
Comment (oesukam) [inline src/Components/Header/Header.tsx]: Why didn't you use the short format of Fragment ?

File: src/Containers/TopLandingSection/index.scss [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.top-landing-section{
```
Comment (oesukam) [inline src/Containers/TopLandingSection/index.scss]: Space before the parentheses to make it clean

**Comment by espoirMur** [comment]
Also, can you please change the app logo and the app title?
I am keep seeing react and the react logo.
But otherwise, it looks nice, good responsiveness...

File: src/Containers/TopLandingSection/index.scss [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+            margin-bottom: 20vh;
+            padding-right: 4vw;
+        }
+        &--description{
+            font-size: 3vh;
+            margin-bottom: 5vh;
+            padding-right: 4vw;
+        }
+    }
+}
```
Comment (oesukam) [inline src/Containers/TopLandingSection/index.scss]: An empty ine shouls be added the end of each files to smothly be treated in different platforms or OS

File: src/Interfaces/Action.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export default interface Action {
+  type: string;
+  payload: any;
```
Comment (oesukam) [inline src/Interfaces/Action.ts]: Avoid using any, be specific

File: src/Pages/Home/index.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+h1{
+    font-size: 5em;
+}
```
Comment (oesukam) [inline src/Pages/Home/index.scss:3]: Empty line required

File: src/Pages/Home/index.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+h1{
```
Comment (oesukam) [inline src/Pages/Home/index.scss:1]: Space after the tag

File: src/helpers/capitalize.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+export default (str: string) => `${str[0].toUpperCase()}${str.substr(1)}`;
```
Comment (oesukam) [inline src/helpers/capitalize.ts:1]: What happens if a string only have one character or it's empty?

File: src/redux/actions/changeName.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { CHANGE_NAME } from "../actionTypes";
+
+export const changeName = (name: string) => (dispatch: Function) => {
+  // Do some actions
```
Comment (oesukam) [inline src/redux/actions/changeName.ts:4]: Unnecessary comment as the action is clear

File: src/redux/initialStates/index.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import name from './name';
+
+export default {
+    name
+};
```
Comment (oesukam) [inline src/redux/initialStates/index.ts:5]: New line required

File: src/redux/initialStates/name.ts [hunk 11]
```diff
@@ -0,0 +1,3 @@
+export default {
+    sample: true,
+}
```
Comment (oesukam) [inline src/redux/initialStates/name.ts]: New line required

File: src/Containers/TopLandingSection/index.scss [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+        }
+        &--slogan{
+            font-size: 4vh;
+            margin-bottom: 20vh;
+            padding-right: 4vw;
+        }
+        &--description{
+            font-size: 3vh;
+            margin-bottom: 5vh;
+            padding-right: 4vw;
```
Comment (dusmel) [inline src/Containers/TopLandingSection/index.scss:15]: I don't think it's a good idea to have font size depending on the size of a screen, and even less the height.

File: src/routes/home.ts [hunk 13]
```diff
@@ -0,0 +1,7 @@
+import Home from "../Pages/Home";
+
+export default {
+    name: 'Home',
+    path:'/',
+    component: Home
+}
```
Comment (oesukam) [inline src/routes/home.ts]: New line required

File: src/routes/index.ts [hunk 14]
```diff
@@ -0,0 +1,7 @@
+import home from "./home";
+import notFound from "./notFound";
+
+export default [
+    home,
+    notFound
+];
```
Comment (oesukam) [inline src/routes/index.ts]: New line required

File: src/routes/notFound.ts [hunk 15]
```diff
@@ -0,0 +1,7 @@
+import NotFound from '../Pages/NotFound';
+
+export default {
+    name: 'Not Found',
+    path: '*',
+    component: NotFound
+}
```
Comment (oesukam) [inline src/routes/notFound.ts]: New line required

**Review by dusmel — COMMENTED** [review]
LGTM!

File: src/style/index.scss [hunk 16]
```diff
@@ -1,3 +1,3 @@
 @tailwind base;
 @tailwind components;
-@tailwind utilities;
+@tailwind utilities;
```
Comment (oesukam) [inline src/style/index.scss]: New line required

**Review by oesukam — CHANGES_REQUESTED** [review]
Great job. I left few comments related to typing validations and code formating

File: src/Containers/TopLandingSection/index.scss (hunk 12 above)
Comment (gracelungu) [reply src/Containers/TopLandingSection/index.scss:15]: @dusmel This is crucial for big screens or with projectors, the text will scale accordingly, making the text much more comfortable. I can use the viewport but it won't be as great as the one above

File: src/Containers/TopLandingSection/index.scss (hunk 4 above)
Comment (gracelungu) [reply src/Containers/TopLandingSection/index.scss]: Cool, but there is a task to help fix these types of errors.

File: src/Interfaces/Action.ts (hunk 5 above)
Comment (gracelungu) [reply src/Interfaces/Action.ts]: @oesukam This is just a boilerplate to give an idea, but I will change it.

File: src/Containers/TopLandingSection/index.scss (hunk 12 above)
Comment (dusmel) [reply src/Containers/TopLandingSection/index.scss:15]: @gracelungu imagine how big the font will be on those vertical screens. Having the viewport meta tag in the head will handle most of this case.

Comment (gracelungu) [reply src/Containers/TopLandingSection/index.scss:15]: @dusmel You can try it and you will see. and keep in mind to  use the right ratios

**Pull request merged**